zebra/rtadv.c

```diff
@@ -1385,6 +1385,76 @@ void zebra_interface_radv_enable(ZAPI_HANDLER_ARGS)
 	zebra_interface_radv_set(client, hdr, msg, zvrf, 1);
 }
 
+static void show_zvrf_rtadv_adv_if_helper(struct vty *vty,
+					  struct adv_if_list_head *adv_if_head)
+{
+	struct adv_if *node = NULL;
+
+	if (!adv_if_is_empty_internal(adv_if_head)) {
+		frr_each (adv_if_list, adv_if_head, node) {
+			vty_out(vty, "    %s\n", node->name);
+		}
+	}
+
+	vty_out(vty, "\n");
+}
+
+static void show_zvrf_rtadv_helper(struct vty *vty, struct zebra_vrf *zvrf)
+{
+	vty_out(vty, "VRF: %s\n", zvrf_name(zvrf));
+	vty_out(vty, "  Interfaces:\n");
+	show_zvrf_rtadv_adv_if_helper(vty, &zvrf->rtadv.adv_if);
+
+	vty_out(vty, "  Interfaces(msec):\n");
+	show_zvrf_rtadv_adv_if_helper(vty, &zvrf->rtadv.adv_msec_if);
+}
+
+DEFPY(show_ipv6_nd_ra_if, show_ipv6_nd_ra_if_cmd,
+      "show ipv6 nd ra-interfaces [vrf<NAME$vrf_name|all$vrf_all>]",
+      SHOW_STR IP6_STR
+      "Neighbor discovery\n"
+      "Route Advertisement Interfaces\n" VRF_FULL_CMD_HELP_STR)
+{
+	struct zebra_vrf *zvrf = NULL;
+
+	if (!vrf_is_backend_netns() && (vrf_name || vrf_all)) {
+		vty_out(vty,
+			"%% VRF subcommand only applicable for netns-based vrfs.\n");
+		return CMD_WARNING;
+	}
```

Comment:
this command should work for netns based commands too.

Reply:
actually, it is more the other way.
I don't see why forcing the user to dump interface infromation on default vrf, whereas the user may be vrf oriented, and may expect that the interface is located behind a give vrf.

Reply:
Reposting what I said in slack for discussion here:


```
that command works for netns

more so actually
the vrf subcommand ONLY works for vrf-netns
since with vrf netns it's all handled in separate namespaces you need that info.
with vrf-lite, it's all in the default vrf, so using "vrf" doesn't do anything

with vrf-lite is doesn't make sense though. RAs are not handled in separate VRFs
this is mostly a debugging command so we can verify the state of the rtadv data

I could make it hidden.
```

I would prefer to not delimit it by VRFs for vrf-lite because that is not reflected in the actual state of the tables this is outputting.